src/commands/apphosting-config-export.ts

```diff
@@ -0,0 +1,64 @@
+import { Command } from "../command";
+import { logger } from "../logger";
+import { Options } from "../options";
+import { needProjectId } from "../projectUtils";
+import { requireAuth } from "../requireAuth";
+import * as secretManager from "../gcp/secretManager";
+import { requirePermissions } from "../requirePermissions";
+import { APPHOSTING_LOCAL_YAML_FILE, discoverProjectRoot } from "../apphosting/config";
+import { fetchSecrets, loadConfigToExport } from "../apphosting/secrets";
+import { join, resolve } from "path";
+import * as fs from "../fsutils";
+import { AppHostingYamlConfig } from "../apphosting/yaml";
+
+export const command = new Command("apphosting:config:export")
+  .description(
+    "Export apphosting configurations such as secrets into an apphosting.local.yaml file",
+  )
+  .option(
+    "-s, --secrets <apphosting.yaml or apphosting.<environment>.yaml file to export secrets from>",
+    "This command combines the base apphosting.yaml with the specified environment-specific file (e.g., apphosting.staging.yaml). If keys conflict, the environment-specific file takes precedence.",
+  )
+  .before(requireAuth)
+  .before(secretManager.ensureApi)
+  .before(requirePermissions, ["secretmanager.versions.access"])
+  .action(async (options: Options) => {
+    const projectId = needProjectId(options);
+    const environmentConfigFile = options.secrets as string | undefined;
+    const cwd = process.cwd();
+
+    // Load apphosting.local.yaml file if it exists. Secrets should be added to the env list in this object and written back to the apphosting.local.yaml
+    let localAppHostingConfig: AppHostingYamlConfig = AppHostingYamlConfig.empty();
+    const projectRoot = discoverProjectRoot(cwd);
+    let localAppHostingConfigPath: string | undefined;
+
+    if (projectRoot) {
+      localAppHostingConfigPath = resolve(projectRoot, APPHOSTING_LOCAL_YAML_FILE);
+      if (fs.fileExistsSync(localAppHostingConfigPath)) {
+        localAppHostingConfig = await AppHostingYamlConfig.loadFromFile(localAppHostingConfigPath);
+      }
+    }
+
+    const configToExport = await loadConfigToExport(cwd, environmentConfigFile);
+    const secretsToExport = configToExport.secrets;
+    if (!secretsToExport) {
+      logger.warn("No secrets found to export in the choosen apphosting files");
+      return;
+    }
+
+    const secretMaterial = await fetchSecrets(projectId, secretsToExport);
+    for (const [key, value] of secretMaterial) {
+      localAppHostingConfig.addEnvironmentVariable({
+        variable: key,
+        value: value,
+        availability: ["RUNTIME"],
+      });
+    }
+
+    // update apphosting.local.yaml
+    localAppHostingConfig.upsertFile(
+      localAppHostingConfigPath ?? join(cwd, APPHOSTING_LOCAL_YAML_FILE),
```

Comment:
should this not upsert at the project root?

Reply:
yep, I was a not sure what to do if the root could not be found (i.e no `apphosting.yaml` is present). This implementation lets them still export secrets even if they only had an environment specific app hosting config file, but I think it makes sense to force users to initialize their apphosing.yaml file and expect it to be there if they're using apphosting.

Reply:
I understand now, it upserts at the original location or writes a new apphosting.local.yaml file at cwd

I think this is an acceptable fall back. Maybe we can make this clear in the console output, eg 

"Exported to local config at /full/path/to/existing/apphosting.local.yaml."

vs

"
Exported to local config at cwd/apphosting.local.yaml.
Make sure this file is stored at your backend root folder (usually next to apphosting.yaml and firebase.json).
"